blog/content/post/2024-04-12-odroid-m1-secure-boot.md

```diff
@@ -0,0 +1,649 @@
+rsa,modulus     rsa,c
+rsa,exponent    rsa,r-squared
+rsa,n0-inverse  rsa,modulus
+rsa,num-bits    rsa,exponent-BN
+key-name-hint   rsa,exponent
+                rsa,n0-inverse
+                rsa,num-bits
+                key-name-hint
+```
+
+To build `mkimage` that can add public key to SPL I had to set
+`CONFIG_FIT_SIGNATURE`. The easiest way to do that is to use `make menuconfig`. I
+also set `CONFIG_SPL_FIT_SIGNATURE`.
+
+```text
+ .config - U-Boot 2017.09 Configuration
+ → Search (signature) ────────────────────────────────────────────────────────
+  ┌──────────────────────────── Search Results ────────────────────────────┐
+  │ Symbol: FIT_SIGNATURE [=n]                                             │
+  │ Type  : boolean                                                        │
+  │ Prompt: Enable signature verification of FIT uImages                   │
+  │   Location:                                                            │
+  │     -> Boot images                                                     │
+  │ (1)   -> Support Flattened Image Tree (FIT [=y])                       │
+  │   Defined at Kconfig:224                                               │
+  │   Depends on: FIT [=y] && DM [=y]                                      │
+  │   Selects: RSA [=y] && CONSOLE_DISABLE_CLI [=n]                        │
+  │                                                                        │
+  │                                                                        │
+  │ Symbol: SPL_FIT_SIGNATURE [=n]                                         │
+  │ Type  : boolean                                                        │
+  │ Prompt: Enable signature verification of FIT firmware within SPL       │
+  │   Location:                                                            │
+  │     -> Boot images                                                     │
+  │ (2)   -> Support Flattened Image Tree (FIT [=y])                       │
+  │   Defined at Kconfig:309                                               │
+  │   Depends on: FIT [=y] && SPL [=y] && SPL_DM [=y]                      │
+  │   Selects: SPL_FIT [=y] && SPL_RSA [=y]                                │
+  ├────────────────────────────────────────────────────────────────( 99%)──┤
+  │                                < Exit >                                │
+  └────────────────────────────────────────────────────────────────────────┘
+```
+
+### Building U-Boot
+
+I used `make.sh` script to build U-Boot. Before setting `FIT_SIGNATURE` and
+`SPL_FIT_SIGNATURE` variables build completed successfully, after setting those
+variables `make.sh` script ends in error but fortunately everything I needed
+was built correctly. Build output should show
+`Platform RK3568 is build OK, with exist .config` message along with error:
+
+* When there are no keys in `u-boot/keys` directory
+
+  ```text
+  ERROR: No keys/dev.key
+  ```
+
+* When there are keys in `u-boot/keys` directory
```

Comment:
Yes I'm sure because that's how I did this. Error only happens after needed files are built I think during signing process. It's not needed because later I do it manually.